envs/example/defaults.yml

```diff
@@ -312,70 +299,69 @@ keystone:
     - name: keystone
       type: identity
       description: 'Identity Service'
-      public_url: https://{{ endpoints.main }}:5001/v2.0
-      internal_url: https://{{ endpoints.main }}:5001/v2.0
-      admin_url: https://{{ endpoints.main }}:35358/v2.0
+      public_url: "{{ endpoints.keystone.url.public }}/{{ endpoints.keystone.version }}"
+      internal_url: "{{ endpoints.keystone.url.internal }}/{{ endpoints.keystone.version }}"
+      admin_url: "{{ endpoints.keystone.url.admin }}/{{ endpoints.keystone.version }}"
     - name: keystonev3
       type: identityv3
       description: 'Identity Service v3'
-      public_url: https://{{ endpoints.main }}:5001/v3
-      internal_url: https://{{ endpoints.main }}:5001/v3
-      admin_url: https://{{ endpoints.main }}:35358/v3
+      public_url: "{{ endpoints.keystone.url.public }}/{{ endpoints.keystonev3.version }}"
+      internal_url: "{{ endpoints.keystone.url.internal }}/{{ endpoints.keystonev3.version }}"
+      admin_url: "{{ endpoints.keystone.url.admin }}/{{ endpoints.keystonev3.version }}"
```

Comment:
Why does keystone have a `version` sub-key, while things like cinder which has multiple versions for endpoints just uses a `path` subkey for each version?